Automattic/wp-calypso
# Track if user scrolls to end of Design Picker

#### Proposed Changes

* ~~Introduce a new hook to track scroll when it reaches the bottom of the page~~
* We decided to take advantage of [useInView](https://github.com/Automattic/wp-calypso/blob/30d5518166fbcc8f40ddc6167fd7ee70cd3c4c62/client/lib/use-in-view/index.ts#L3) to track instead of onscroll
* Trigger event `calypso_signup_design_scrolled_to_end` with 2 props `intent` and `site_category` when unified design picker page is scrolled to bottom

#### Testing Instructions



* Set up your browser so you can easily see Tracks events -- I would strongly recommend the [Tracks Vigilante Chrome extension](https://github.com/Automattic/tracks-chrome-extension)
* Open the Calypso live branch for this PR (or via your local Calypso environment)
* Navigate to the Design Picker step -- you can navigate directly there via `/setup/site-setup/designSetup?siteSlug=${ siteSlug }`
* Inspect the Tracks events logged for `calypso_signup_design_scrolled_to_end`. Also verify that the events have valid properties
* Confirm that the tracks event only fired once when category changed
![BN7nsA.png](https://user-images.githubusercontent.com/10071857/207592529-b8e3ff28-af46-4e0b-a337-3d49340c44ae.png)
Recording:

https://user-images.githubusercontent.com/10071857/207592582-0a99513f-f8e7-4686-a874-7b6daa81df5b.mp4


#### Pre-merge Checklist





Related to [#1104](https://github.com/Automattic/dotcom-forge/issues/1104)
